tools/depcheck/src/allowlist.rs

```diff
@@ -142,13 +142,18 @@ pub const EXTRA_DATAGEN_DEPS: &[&str] = &[
     "databake-derive",
     "elsa",
     "erased-serde",
+    "fraction",
```

Comment:
I know this conversation has been going back and forth for a little while, but I do share some concerns with @robertbastian about the `fraction` crate. We already pull in the `num-rational` crate which appears to support the same sort of abstraction: `pub type BigRational = Ratio<BigInt>;`. I don't understand why we need the extra crate. Besides this, `fraction` has less than 6% of the lifetime downloads of `num-rational`.

This isn't blocking because this is only datagen but it will definitely be a concern when we put this into the library crate.

Reply:
I've already explained to Robert the reasons for using the `GenericFraction` change:

- **Sign Separation**: Unlike `BigRational`, which includes the sign in the numerator, `GenericFraction` allows for a clear separation between the numerator/denominator and the sign.
- **Byte Sequence Extraction**: We need to convert `BigInt` to `BigUint` to extract the byte sequences for both the numerator and denominator.
- **Decimal String Creation**: `GenericFraction` supports initialization from decimal strings.

Given these reasons, `GenericFraction` is more appropriate for use in the datagen. However, for calculations during the conversion process, `BigRational` might be the better choice.

Reply:
Hm. All of these seem to me to be relatively easily solved with some code. I don't think it is the right choice to depend on a large crate with a bunch of unsafe for this.

I'm comfortable adding the `num` deps, since those are an ecosystem standard.

Reply:
Looking at the current code; it doesn't look like we rely on Sign Separation, i'm not sure where the byte sequence extraction is relevant, and decimal string creation we only use to parse a `BigUint`. We can do that directly.

Reply:
mmmm .. in this case, I can use `BigRational`

Reply:
but I am still not 100% convinced

Reply:
I'm already making the switch. It's turning out to be quite straightforward.